client/blocks/location-search/index.jsx

```diff
@@ -0,0 +1,98 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { loadScript } from 'lib/load-script';
+import config from 'config';
+
+/**
+ * Internal dependencies
+ */
+import CompactCard from 'components/card/compact';
+import SearchCard from 'components/search-card';
+
+let autocompleteService = null;
+
+class LocationSearch extends Component {
+	static propTypes = {
+		onPredictionClick: PropTypes.func,
+		predictionHref: PropTypes.string,
+	};
+
+	state = { predictions: [] };
+
+	componentDidMount() {
+		if ( ! autocompleteService ) {
+			autocompleteService = {};
+			loadScript(
+				`//maps.googleapis.com/maps/api/js?key=${ config(
+					'google_maps_api_key'
+				) }&libraries=places`,
+				function() {
+					// eslint-disable-next-line no-undef
+					autocompleteService = new google.maps.places.AutocompleteService();
+				}
+			);
+		}
+	}
+
+	updatePredictions = predictions => {
+		this.setState( { predictions } );
+	};
+
+	handleSearch = query => {
+		this.setState( {
+			query: query,
+		} );
+
+		if ( query ) {
+			autocompleteService.getPlacePredictions(
+				{
+					input: query,
+					types: [ 'establishment' ],
+				},
+				this.updatePredictions
+			);
+		} else {
+			this.updatePredictions( [] );
+		}
+	};
+
+	renderPrediction = prediction => {
+		const { predictionHref, onPredictionClick } = this.props;
+
+		return (
+			<CompactCard
+				key={ prediction.place_id }
+				href={ predictionHref }
+				onClick={ onPredictionClick }
```

Comment:
How do you plan to retrieve the `place_id` with this event handler? I would avoid accessing the DOM from this handler to retrieve the `key` attribute (I'm not even sure it's part of the DOM).